dotCMS/src/main/java/com/dotcms/csspreproc/dartsass/CompilerOptions.java

```diff
@@ -0,0 +1,323 @@
+package com.dotcms.csspreproc.dartsass;
+
+import com.dotmarketing.util.Config;
+import io.vavr.Lazy;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class allows you to enable/disable different configuration parameters that define the behavior of the Dart SASS
+ * Compiler. If no parameters are changed upon instantiation, the default dotCMS-specific configuration will be used
+ * instead. Here's the list of parameters that can be activated for the compiler, which can be found in the {@code
+ * dotmarketing-config.properties} file:
+ * <ul>
+ *     <li>{@code Verbose}: ({@code false} by default) Prints all deprecation warnings even when they're repetitive
+ *     .</li>
+ *     <li>{@code Expanded CSS}: ({@code true} by default) Controls the output style of the resulting CSS. Dart Sass
+ *     supports two output styles: {@code expanded} (the default) writes each selector and declaration on its own
+ *     line. And {@code compressed} removes as many extra characters as possible, and writes the entire stylesheet on
+ *     a single line.</li>
+ *     <li>{@code Error in CSS}: ({@code true} by default) This flag tells Sass whether to emit a CSS file when an
+ *     error occurs during compilation. This CSS file describes the error in a comment and in the "content" property
+ *     of "body::before", so that you can see the error message in the browser without needing to switch back to the
+ *     terminal.</li>
+ *     <li>{@code Stop on Error}: ({@code true} by default) This flag tells Sass to stop compiling immediately when
+ *     an error is detected, rather than trying to compile other Sass files that may not contain errors.</li>
+ *     <li>{@code Deprecation Warnings}: ({@code false} by default) This flag tells Sass to emit any warnings when
+ *     compiling. By default, Sass emits warnings when deprecated features are used or when the {@code @warn} rule is
+ *     encountered.</li>
+ *     <li>{@code Deprecation Warnings From Dependencies}: ({@code false} by default) This flag tells Sass not to
+ *     emit deprecation warnings that come from dependencies. It considers any file that’s transitively imported
+ *     through a load path to be a "dependency".</li>
+ * </ul>
+ *
+ * @author Jose Castro
+ * @since Jul 25th, 2022
+ */
+public class CompilerOptions implements Serializable {
+
+    private final boolean verbose;
+    private final boolean expandedCss;
+    private final boolean errorInCss;
+    private final boolean stopOnError;
+    private final boolean deprecationWarnings;
+    private final boolean deprecationWarningsFromDependencies;
+
+    private static final String VERBOSE = "dartsass.compiler.verbose";
```

Comment:
Maybe this should be part as an attribute if the `SassCommands` enum defined below.

Reply:
Done!